matching/src/Pattern.hs

```diff
@@ -120,7 +121,7 @@ swap ix tm = Fix (Swap ix tm)
 -- [ Matrix ]
```

Comment:
These matrix functions are pretty difficult to read and / or reason about. How about some tests?